reconcile/external_resources/manager.py

```diff
@@ -133,6 +133,40 @@ def __init__(
         self.thread_pool_size = thread_pool_size
         self.dry_runs_validator = dry_runs_validator
 
+    def _resource_spec_changed(
+        self, reconciliation: Reconciliation, state: ExternalResourceState
+    ) -> bool:
+        return reconciliation.resource_hash != state.reconciliation.resource_hash
+
+    def _resource_drift_detection_ttl_expired(
+        self, reconciliation: Reconciliation, state: ExternalResourceState
+    ) -> bool:
+        return (datetime.now(state.ts.tzinfo) - state.ts).total_seconds() > (
+            reconciliation.module_configuration.reconcile_drift_interval_minutes * 60
+        )
+
+    def _reconciliation_module_config_overridden(
+        self, reconciliation: Reconciliation, state: ExternalResourceState
+    ) -> bool:
+        return reconciliation.module_configuration.overridden and (
+            reconciliation.module_configuration.image_version
+            != state.reconciliation.module_configuration.image_version
+        )
```

Comment:
should image version overridden be a special case, or we should trigger reconcile whenever new desired image_version not same as last run image version in `state`?

Reply:
This will trigger a reconciliation for all the resources with every bump, which is not desired right now. We can consider it once we have rolling updates at account level.